build_tools/configure_ci.py

```diff
@@ -174,37 +180,100 @@ def should_ci_run_given_modified_paths(paths: Optional[Iterable[str]]) -> bool:
 
 
 # --------------------------------------------------------------------------- #
-# Working with pull request labels
+# Matrix creation logic based on PR, push or workflow_dispatch
 # --------------------------------------------------------------------------- #
 
+amdgpu_family_info_matrix = {
+    "gfx94X": {
+        "linux": {"runs-on": "linux-mi300-1gpu-ossci-rocm", "target": "gfx94X-dcgpu"}
+    }
+}
+
 
-def get_pr_labels() -> List[str]:
+def get_pr_labels(args) -> List[str]:
     """Gets a list of labels applied to a pull request."""
-    labels = json.loads(os.environ.get("PR_LABELS", "[]"))
+    labels = json.loads(args.get("pr_labels"))
     return labels
 
 
+def matrix_generator(is_pull_request, is_workflow_dispatch, is_push, args):
```

Comment:
Add logging for what this is doing. Current logs aren't showing much: https://github.com/ROCm/TheRock/actions/runs/14133522224/job/39599600974?pr=310#step:3:14